src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -362,25 +364,72 @@ public Info info() throws DockerException, InterruptedException {
     WebTarget resource = resource()
         .path("containers").path("json");
 
+    final Map<String, List<String>> filters = newHashMap();
     for (ListContainersParam param : params) {
-      resource = resource.queryParam(param.name(), param.value());
+      if (param instanceof ListContainersFilterParam) {
```

Comment:
I considered something like that. Making one method write out filter params for images and containers would avoid a lot of duplicate code between `listContainers` and `listImages`. But the inheritance of the classes you would pass in to it seems... complicated. 

To have one method handle both types, it would have to accept a generic `ListParam`, and `ListImagesParam` and `ListContainersParam` would have to be modified to inherit from `ListParam`. That wouldn't be hard.

The part that confuses me is how to handle the `List*FilterParam`s. Right now they are subclasses of the respective non-`Filter` classes. But if we made a generic `ListParam`, it seems we would need to make a generic `ListFilterParam` subclass. But `ListImagesFilterParam` can't be both a `ListImagesParam` and a `ListFilterParam`. So which would it be? How would the methods in `ListImagesParam` and `ListContainersParam` need to be changed to support the new `List*FilterParam` classes?

I'm not saying there aren't good answers for these questions, just saying that it may be a little tricky to get right. (Or maybe it's super simple and I'm missing something. Could be that too.)